component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -525,6 +523,65 @@ private String getParameterFromURIString(String uriString, String parameterName)
         }
         return StringUtils.EMPTY;
     }
-}
 
+    /**
+     * Resolves the callback URL from the authenticator properties. If a callback URL is provided, that value is
+     * returned. If empty, returns the global commonauth endpoint when organization-specific commonauth is disabled,
+     * or otherwise the organization-specific commonauth endpoint.
+     *
+     * @param authenticatorProperties the authenticator properties map.
+     * @return the resolved callback URL.
+     * @throws RuntimeException if building the callback URL fails.
+     */
+    @Override
+    protected String getCallbackUrl(Map<String, String> authenticatorProperties) {
+
+        String callbackUrl = authenticatorProperties.get(Office365AuthenticatorConstants.CALLBACK_URL);
+        if (StringUtils.isNotEmpty(callbackUrl)) {
+            return callbackUrl;
+        }
 
+        if (!isOrgSpecificCommonAuthURLEnabled()) {
+            return IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, true, true);
+        }
+
+        return buildOrgSpecificCommonAuthURL();
+    }
+
+    /**
+     * Returns whether the organization-specific commonauth endpoint is enabled in the authenticator parameters.
+     *
+     * @return true if organization-specific commonauth is enabled, false otherwise.
+     */
+    private boolean isOrgSpecificCommonAuthURLEnabled() {
+
+        if (getAuthenticatorConfig() == null || getAuthenticatorConfig().getParameterMap() == null) {
+            return false;
+        }
+
+        String value = getAuthenticatorConfig()
+                .getParameterMap()
+                .get(Office365AuthenticatorConstants.USE_ORG_SPECIFIC_COMMON_AUTH_URL);
+
+        return StringUtils.isNotBlank(value) && Boolean.parseBoolean(value);
```

Comment:
Boolean.parseBoolean(value) will handle the null case as well hence StringUtils.isNotBlank(value) is not needed

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/38/commits/b8dc6f240d713be3f4608845c7333f65d7d587cc.